tutorials/ui/gui_using_fonts.rst

```diff
@@ -8,51 +8,634 @@ Godot allows you to set specific fonts for different UI nodes.
+distance field texture. This means caching can be done purely on a per-font
+basis, without taking the font size into consideration. However, the initial
+rendering of MSDF fonts is slower compared to a traditional rasterized font at a
+medium size.
+
+To avoid stuttering issues related to font rendering, it is possible to
+*prerender* certain glyphs. This can be done for all glyphs you intend to use
+(for optimal results), or only for common glyphs that are most likely to appear
+during gameplay (to reduce file size). Glyphs that aren't pre-rendered will be
+rasterizd on-the-fly as usual.
+
+.. note::
+
+    In both cases (traditional and MSDF), font rasterization is done on the CPU.
+    This means that the GPU performance doesn't affect how long it takes for fonts
+    to be rasterized.
+
+Open the Advanced Import Settings dialog by double-clicking the font file in the
+FileSystem dock. You can also select the font in the FileSystem dock, go to the
+Import dock then choose **Advanced…** at the bottom:
+
+.. figure:: img/using_fonts_advanced_import_settings.webp
+   :align: center
+
+   Import dock
+
+Move to the **Pre-render Configurations** tab of the Advanced Import Settings dialog,
+then add a configuration by clicking the "plus" symbol:
+
+.. figure:: img/using_fonts_advanced_import_settings_prerender_new_configuration.webp
+   :align: center
+   :alt: Adding a new prerendering configuration in the Advanced Import Settings dialog
+
+   Adding a new prerendering configuration in the Advanced Import Settings dialog
+
+After adding a configuration, make sure it is selected by clicking its name
+once. You can also rename the configuration by double-clicking it.
+
+There are 2 ways to add glyphs to be prerendered to a given configuration. It is
+possible to use both approaches in a cumulative manner:
+
+**Using preexisting text**
+
+While it requires manually specifying text that will appear in the game, this is
+the most efficient approach for games which don't feature user text input. This
+approach is worth exploring for mobile games to reduce the file size of the
+distributed app.
+
+To use existing text as a baseline for prerendering, go to the **Glyphs from the
+Text** sub-tab of the Advanced Import Settings dialog, enter text in the window
+on the right, then click **Shape Text and Add Glyphs** at the bottom of the
+dialog:
+
+.. figure:: img/using_fonts_advanced_import_settings_prerender_text.webp
+   :align: center
+   :alt: Enabling prerendering in the Advanced Import Settings dialog, Glyphs from the Text tab
+
+   Enabling prerendering in the Advanced Import Settings dialog with the **Glyphs from the Text** tab
+
+.. tip::
+
+    If your project supports `internationalization
+    <doc_internationalizing_games>`_, you can paste the contents of your CSV or
+    PO files in the above box to quickly prerender all possible characters that
+    will be rendered during gameplay.
```

Comment:
This makes me think we should have a button to shape text from all translations in the "Glyphs from the Text" tab. This would be more convenient when using PO translations, or CSV translations split on a per-language basis.

@bruvzg What do you think? I can open a proposal for this if needed.

Reply:
I guess it would make sense to auto process all text (maybe with translation selector and on its own tab).
`Glyphs from the Text` mainly exist to pre-render text with specific OpenType features enabled, not for a batch text processing (since it's impossible to select stylistic variants for the character map, they do not have unique char codes).

Reply:
https://github.com/godotengine/godot/pull/68448